datalad_hirni/commands/import_additional_data.py

```diff
@@ -0,0 +1,168 @@
+
+
+def _add_to_spec(spec, path, meta):
+
+    return spec.append({
+        'type': 'additional',
+        'status': None,  # TODO: process state convention; flags
+        'location': op.relpath(meta['path'], path),
+        'dataset_id': meta['dsid'],
+        'dataset_refcommit': meta['refcommit'],
+        'converter': {'value': None, 'approved': False}
+    })
+
+
+@build_doc
+class ImportAdditionalData(Interface):
+    """
+    """
+
+    _params_ = dict(
+        dataset=Parameter(
+            args=("-d", "--dataset"),
+            metavar='PATH',
+            doc="""specify the dataset to import the DICOM archive into.  If
+            no dataset is given, an attempt is made to identify the dataset
+            based on the current working directory and/or the `path` given""",
+            constraints=EnsureDataset() | EnsureNone()),
+        path=Parameter(
+            args=("path",),
+            metavar='PATH',
+            doc="""path(s) of the data to be imported.""",
+            nargs="*",
+            constraints=EnsureStr()),
+        acqid=Parameter(
+            args=("acqid",),
+            metavar="ACQUISITION ID",
+            doc="""acquisition identifier for the imported data files""",
+            constraints=EnsureStr() | EnsureNone()),
+        target_dir=Parameter(
+            args=("-t", "--target-dir"),
+            metavar="TARGET",
+            doc="""subdirectory of the acquisition to store the data in""",
+            constraints=EnsureStr() | EnsureNone()),
+
+    )
+
+    @staticmethod
+    @datasetmethod(name='hirni_import_data')
+    @eval_results
+    def __call__(acqid, path, dataset=None, target_dir=None):
+
+        ds = require_dataset(dataset, check_installed=True,
+                             purpose="import additional data")
+        acq_dir = op.join(ds.path, acqid)
+        if not op.exists(acq_dir):
+            raise ValueError("Acquisition %s does not yet exist" % acqid)
+
+        path = assure_list(path)
+
+        # TODO: spec file specifiable or fixed path?
+        #       if we want the former, what we actually need is an association
+        #       of acquisition and its spec path
+        #       => prob. not an option but a config
+        spec_path = op.join(ds.path, acqid, "studyspec.json")
+        spec = [r for r in json_py.load_stream(spec_path)] \
+            if op.exists(spec_path) else list()
+
+        ds_meta = ds.metadata(reporton='datasets', return_type='generator',
+                              result_renderer='disabled')
+
+        for p in path:
+
+            src = RI(p)
+            if isinstance(src, PathRI):
+                src = RI("file://" + PathRI(op.abspath(p)).posixpath)  # TODO: Is this the right way for file-scheme on windows?
+
+            dst = op.join(acq_dir, target_dir if target_dir else "",
+                          op.basename(src.path))  # TODO: Again windows: If an actual URL (POSIX-path), does it still work with os.path.basename?
```

Comment:
Please check https://github.com/datalad/datalad/issues/2546 Windows is more clever than people give it credit for. Just the shell is dumb as f...

Reply:
More specifically, we should test whether we can use front-slashes on windows in all cases (also a unit test).

I.e. `C:/users/stupid` instead of 'C:\\users\\stupid'

Reply:
Generally agree, but this actually is about `os.path` module rather than windows itself.
The command should be able to take a URL for the `PATH` argument. If that is used we have some `https://somewhe.re/not/important.txt`. This line would now pass `/not/important.txt` into `ntpath.basename`, which probably is expecting backslashes. But as the comment says - I'm not sure whether it can deal with it.

Reply:
And that is why this is important, and one and the same issue IMHO. If we know that Windows can deal with forward slashes, we should test what ntpath does with them.

Reply:
Here is the relevant test: https://github.com/datalad/datalad/pull/2653

Reply:
Nice.

BTW:
```
>>> import ntpath
>>> ntpath.basename("C:/some/file")
'file'
```

So, what I was concerned about works.